src/compiler/transformers/classFields.ts

```diff
@@ -411,8 +449,89 @@ namespace ts {
             return statement;
         }
 
-        function visitComputedPropertyName(node: ComputedPropertyName) {
-            let expression = visitNode(node.expression, visitor, isExpression);
+        function visitVariableDeclaration(node: VariableDeclaration) {
+            // 14.3.1.2 RS: Evaluation
+            //   LexicalBinding : BindingIdentifier Initializer
+            //     ...
+            //     3. If IsAnonymousFunctionDefinition(|Initializer|) is *true*, then
+            //        a. Let _value_ be ? NamedEvaluation of |Initializer| with argument _bindingId_.
+            //     ...
+            //
+            // 14.3.2.1 RS: Evaluation
+            //   VariableDeclaration : BindingIdentifier Initializer
+            //     ...
+            //     3. If IsAnonymousFunctionDefinition(|Initializer|) is *true*, then
+            //        a. Let _value_ be ? NamedEvaluation of |Initializer| with argument _bindingId_.
+            //     ...
+
+            if (isIdentifier(node.name) && node.initializer &&

```

Comment:
this code is almost the same as visitBindingElement. Is it possible to share code between them?

Reply:
I think I can extract some of the repeated code out related to creating the string literal for the assigned name, since that logic is used in 5 different places in the file.